File: src/models/discussion-message.ts
import { Attachment } from "./attachment";
import { DiscussionMessageRecipient } from "./discussion-message-recipient";

export type DiscussionMessage = Readonly<{
  id: string
  content: string
  creationDate: Date
  /**
   * When undefined, the author is the user itself.
   */
  author?: DiscussionMessageRecipient
  /**
   * When undefined, the receiver is the user itself.
   */
  receiver?: DiscussionMessageRecipient
  partialVisibility: boolean
  amountOfRecipients: number
  files: Attachment[]
}>;

File: library/src/commonMain/kotlin/ink/literate/pawnote/models/DiscussionMessage.kt
package ink.literate.pawnote.models

import kotlinx.datetime.LocalDateTime

data class DiscussionMessage<T>(
    val id: String,
    val content: String,
    val creationDate: LocalDateTime,
    /** When undefined, the author is the user itself. */
    val author: DiscussionMessageRecipient? = null,
    /** When undefined, the receiver is the user itself. */
    val receiver: DiscussionMessageRecipient? = null,
    val partialVisibility: Boolean,
    val amountOfRecipients: Int,
    val files: List<Attachment>,
    val data: T
)
